### PR TITLE
Reword and reorder system basics

### PR DESCRIPTION
**Pull request by mbrookes**
For discussion, as I'm still not 100% happy with the order and nesting, and have a couple of questions. (I might also have unintentionally changed some intended meaning - let me know.)

I also removed the code diff, sorry. It was adding noise (hard to follow), and making that section disjointed (lump of abstract code between related clauses).

### CODE REVIEW

File: docs/src/pages/system/basics/basics.md [hunk 1]
```diff
@@ -1,87 +1,36 @@
-# Material-UI System
+# Material-UI style system
```
Comment (oliviertassinari) [inline docs/src/pages/system/basics/basics.md:1]: I think we should treat it as a product name, like Material-UI X, Material-UI Admin, Material-UI Blocks, etc.

```suggestion
# Material-UI System
```

Comment (mbrookes) [reply docs/src/pages/system/basics/basics.md:1]: > I think we should treat it as a product name

I understand your attachment, but two years with no meaningful traction suggests otherwise.

Comment (oliviertassinari) [reply docs/src/pages/system/basics/basics.md:1]: What's the correlation between the two events?

Comment (oliviertassinari) [reply docs/src/pages/system/basics/basics.md:1]: I would explain no traction with a simple argument: no exposure in the docs. Why no exposure in the docs? Because we couldn't use it ourselves.

Comment (mbrookes) [reply docs/src/pages/system/basics/basics.md:1]: > I think we should treat it as a product name

Let me ask a different question: Why?

> What's the correlation relationship between the two events?

What two events? The system package has been available on npm, and publicised in the docs, for two years. Making it a "brand", or something known and used outside of the Material-UI eco system hasn't worked. It didn't worked for `@material-ui/styles` either, so I don't think it's down to exposure.

> no exposure in the docs

It's in the root of the nav.

> making choices that reduce the likeliness of it to work

Such as calling it "System"? 🙂 It's too vague. What kind of system, for what purpose? If you'd prefer to keep "Style System" capitalised, we can try that. I don't see what we gain by it, but it's harmless enough.

**Comment by mbrookes** [comment]
The diff is too big to fix.

**Comment by oliviertassinari** [comment]
I'm done with https://github.com/mui-org/material-ui/pull/23294. I have tried to bring most of your changes into it.
